File: legacy-annotations/src/main/java/com/siimkinks/sqlitemagic/annotation/ViewQuery.java
package com.siimkinks.sqlitemagic.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * <p>
 * Marks field value to be {@link View} defining query.
 * </p>
 * Each view must only have one query.
 */
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.CLASS)
public @interface ViewQuery {
}

File: annotations/src/main/kotlin/com/siimkinks/sqlitemagic/annotation/ViewQuery.kt
package com.siimkinks.sqlitemagic.annotation

import kotlin.annotation.AnnotationRetention.BINARY
import kotlin.annotation.AnnotationTarget.FIELD

/**
 * Marks a `CompiledSelect` value as the defining query for a [View].
 *
 * Each view must have exactly one query declared directly within the view. In Kotlin, declare it
 * as a non-private companion-object property; `@JvmField` is not required. In Java, declare it as
 * an accessible static field.
 */
@Target(FIELD)
@Retention(BINARY)
annotation class ViewQuery
